textx/metamodel.py

```diff
@@ -179,6 +186,16 @@ def __init__(self, file_name=None, classes=None, builtins=None,
         # constructed from string.
         self._enter_namespace(self._namespace_for_file_name(file_name))
 
+    def _enable_global_model_repository(self):
+        """
+        should be called before loading models
+        """
+        from textx.scoping import GlobalModelRepository
+        self._tx_model_repository = GlobalModelRepository()
+
+    def register_scope_provider(self, sp):
```

Comment:
Maybe this should be in plural as there can be more than one scope provider -> `register_scope_providers`

Reply:
sound good. done.